From 327059ff79d902a20b29f24fab292ae1dac81663 Mon Sep 17 00:00:00 2001
From: Baha Mesleh <baha.mesleh@nokia.com>
Date: Wed, 13 Oct 2021 14:53:05 +0000
Subject: [PATCH 11/11] octeontx2-bphy-netdev: limit cpri error logging rate

commit 3e48bb85f68d13c372c135db4726e4ff17eb6bb5 from
git@git.assembla.com:cavium/WindRiver.linux.git

The patch modifies netdev to use net_err_ratelimited API to
have some control over the logs going to journal, otherwise
the log rotation feature deletes the old logs that are relavant
to debugging the original issue in some cases.

Change-Id: If96d036bdd7c5e798a3d391e5c89cfef2da98948
Signed-off-by: Baha Mesleh <baha.mesleh@nokia.com>
Signed-off-by: Naveen Mamindlapalli <naveenm@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/64450
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 .../marvell/octeontx2/bphy/otx2_cpri.c        | 23 +++++++++----------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
index a9b819bac99c..536db19eb257 100644
--- a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
+++ b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
@@ -161,8 +161,8 @@ static int otx2_cpri_process_rx_pkts(struct otx2_cpri_ndev_priv *priv,
 		wqe = (struct cpri_pkt_ul_wqe_hdr *)pkt_buf;
 		netdev = otx2_cpri_get_netdev(wqe->mhab_id, wqe->lane_id);
 		if (unlikely(!netdev)) {
-			pr_err("CPRI Rx netdev not found, cpri%d lmac%d\n",
-			       wqe->mhab_id, wqe->lane_id);
+			net_err_ratelimited("CPRI Rx netdev not found, cpri%d lmac%d\n",
+					    wqe->mhab_id, wqe->lane_id);
 			priv->stats.rx_dropped++;
 			priv->last_rx_dropped_jiffies = jiffies;
 			processed_pkts++;
@@ -170,20 +170,19 @@ static int otx2_cpri_process_rx_pkts(struct otx2_cpri_ndev_priv *priv,
 		}
 		priv2 = netdev_priv(netdev);
 		if (wqe->fcserr || wqe->rsp_ferr || wqe->rsp_nferr) {
-			netif_err(priv2, rx_err, netdev,
-				  "CPRI Rx err,cpri%d lmac%d sw_rd_ptr=%d\n",
-				  wqe->mhab_id, wqe->lane_id,
-				  ul_cfg->sw_rd_ptr);
+			net_err_ratelimited("%s: CPRI Rx err,cpri%d lmac%d sw_rd_ptr=%d\n",
+					    netdev->name,
+					    wqe->mhab_id, wqe->lane_id,
+					    ul_cfg->sw_rd_ptr);
 			priv2->stats.rx_dropped++;
 			priv2->last_rx_dropped_jiffies = jiffies;
 			processed_pkts++;
 			continue;
 		}
 		if (unlikely(!netif_carrier_ok(netdev))) {
-			netif_err(priv2, rx_err, netdev,
-				  "%s {cpri%d lmac%d} link down, drop pkt\n",
-				  netdev->name, priv2->cpri_num,
-				  priv2->lmac_id);
+			net_err_ratelimited("%s {cpri%d lmac%d} link down, drop pkt\n",
+					    netdev->name, priv2->cpri_num,
+					    priv2->lmac_id);
 			priv2->stats.rx_dropped++;
 			priv2->last_rx_dropped_jiffies = jiffies;
 			processed_pkts++;
@@ -203,8 +202,8 @@ static int otx2_cpri_process_rx_pkts(struct otx2_cpri_ndev_priv *priv,
 
 		skb = netdev_alloc_skb_ip_align(netdev, len);
 		if (!skb) {
-			netif_err(priv2, rx_err, netdev,
-				  "CPRI Rx: alloc skb failed\n");
+			net_err_ratelimited("%s:CPRI Rx: alloc skb failed\n",
+					    netdev->name);
 			priv->stats.rx_dropped++;
 			priv->last_rx_dropped_jiffies = jiffies;
 			processed_pkts++;
-- 
2.31.1

